lib/src/https.rs

```diff
@@ -1960,6 +2071,121 @@ impl Proxy {
                 .borrow_mut()
                 .remove_custom_answer(cluster_id);
         }
+
+        Ok(None)
+    }
+
+    pub fn add_https_frontend(
+        &mut self,
+        front: HttpFrontend,
+    ) -> anyhow::Result<Option<ProxyResponseContent>> {
+        match self
+            .listeners
+            .values()
+            .find(|l| l.borrow().address == front.address)
+        {
+            Some(listener) => {
+                let mut owned = listener.borrow_mut();
+                owned.set_tags(front.hostname.to_owned(), front.tags.to_owned());
+                owned.add_https_front(front);
+                Ok(None)
+            }
+            None => bail!("adding front {:?} to unknown listener", front),
+        }
+    }
+
+    pub fn remove_https_frontend(
+        &mut self,
+        front: HttpFrontend,
+    ) -> anyhow::Result<Option<ProxyResponseContent>> {
+        match self
+            .listeners
+            .values()
+            .find(|l| l.borrow_mut().address == front.address)
+        {
+            Some(listener) => {
+                let mut owned = listener.borrow_mut();
+                owned.set_tags(front.hostname.to_owned(), None);
+                owned.remove_https_front(front);
+                Ok(None)
+            }
+            None => bail!("No listener found for the frontend {:?}", front),
```

Comment:
Same as before, to me the semantic is good even if there is no listener, so we could remove this returns `Ok(())` and use an `if let Some(..) = ..`